### ISSUE TITLE
MinkowskiNet not learning anything on my 3D dataset

### ISSUE THREAD

**Issue by aniqueakhtar** [StanfordVL/MinkowskiEngine#23]
I have labeled 3D point cloud dataset much similar to ScanNet. I only have 3D points and the intensity value of that point. So I am feeding the intensity as a feature and the 3D points as the coordinates.
Here's the code I am using to run the network:
```
device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
# Define a model and load the weights
model = MinkUNet34C(1, data.N_CLASSES).to(device)

training_epochs=512
optimizer = optim.Adam(model.parameters())

for epoch in range(training_epoch, training_epochs+1):
    model.train()
    start = time.time()
    train_loss=0
    for i,batch in enumerate(data.train_data_loader):
        start2 = time.time()
        # Get new data
        coords = batch['x'][0]
        feat = batch['x'][1]
        label = batch['y']
        inputs = ME.SparseTensor(feat, coords=coords).to(device)
        label = label.to(device)
        
        # Forward
        output = model(inputs)        
        predictions = output.F
        # Measure loss
        loss = torch.nn.functional.cross_entropy(predictions,label)
        train_loss+=loss.item()
        # Back propagate
        loss.backward()
        optimizer.step()
        del loss
        del predictions
        del output
        if i%100 == 0:
            log_string('Batch_processed = %03d / %03d,  time = %f' % ( i, data.len_train//data.batch_size, time.time()-start2))
    log_string('%03d  Train loss %f,  time = %f s' % (epoch, train_loss/(i+1), time.time() - start))
    del train_loss
```

Not to mention that this is pretty slow. However, even after waiting a few days, it is clear that the network is not learning anything.

**Comment by chrischoy** [comment]
Hi, sorry for the late reply.

I created a complete training example on a dummy dataset last night and put it on https://github.com/StanfordVL/MinkowskiEngine/blob/master/examples/two_dim_training.py.

The network correctly trains and predicts as expected.

Could you see if you made some mistakes? Also, I made some changes that reverts back to the original CUDA kernels in v0.2.4 and it is now reasonably fast.

**Comment by chrischoy** [comment]
Also, here's the tutorial https://stanfordvl.github.io/MinkowskiEngine/demo/training.html#

**Comment by chrischoy** [comment]
Closing it for no updates.

**Comment by aniqueakhtar** [comment]
I went through the example line by line. My network still seems to be learning nothing. I printed the outputs and the inputs from my network and the example here. I am feeding exactly the same thing and still not getting any convergence.
Maybe share the training code for ScanNet and I can replace ScanNet dataset with my own to see what's going on here?
Thank you

**Comment by aniqueakhtar** [comment]
What I have also noticed while printing the output of the network is most of the classes output is in negative except for 3 classes. The IOU on these classes comes up to 50% which is pretty reasonable too, but I fail to see why the rest of the outputs are negative small numbers. Overall the loss function decreases over time but not enough decrease and it is only giving me answers for three classes out of the 18 classes.
This is what the network output is like:
<img width="484" alt="Screen Shot 2019-09-05 at 1 12 00 PM" src="https://user-images.githubusercontent.com/2750083/64377232-d5502700-cfde-11e9-8b8d-ddc437d21777.png">
